cubi_tk/sodar/ingest.py

```diff
@@ -0,0 +1,366 @@
+                paths = abspath.glob("**/*" if self.args.recursive else "*")
+                for p in paths:
+                    if p.is_file() and not p.suffix.lower() == ".md5":
+                        output_paths.append({"spath": p, "ipath": p.relative_to(abspath)})
+            else:
+                output_paths.append({"spath": src, "ipath": Path(src.name)})
+        return output_paths
+
+    def process_md5_sums(self, paths):
+        """Check input paths for corresponding .md5 file. Generate one if missing."""
+
+        output_paths = paths.copy()
+        for file in paths:
+            p = file["spath"]
+            i = file["ipath"].parent / (p.name + ".md5")
+            upper_path = p.parent / (p.name + ".MD5")
+            lower_path = p.parent / (p.name + ".md5")
+            if upper_path.exists() and lower_path.exists():
+                logger.info(
+                    f"Found both {upper_path.name} and {lower_path.name} in {str(p.parent)}/"
+                )
+                logger.info("Removing upper case version.")
+                upper_path.unlink()
+                output_paths.append({"spath": lower_path, "ipath": i})
+            elif upper_path.exists():
+                logger.info(f"Found {upper_path.name} in {str(p.parent)}/")
+                logger.info("Renaming to " + lower_path.name)
+                upper_path.rename(upper_path.with_suffix(".md5"))
+                output_paths.append({"spath": lower_path, "ipath": i})
+            elif lower_path.exists():
+                output_paths.append({"spath": lower_path, "ipath": i})
+
+            if not lower_path.exists() and self.args.checksums:
+                with lower_path.open("w", encoding="utf-8") as file:
+                    file.write(f"{compute_md5_checksum(p)}  {p.name}")
+                output_paths.append({"spath": lower_path, "ipath": i})
+
+        return output_paths
+
+    def build_jobs(self, source_paths) -> typing.Tuple[TransferJob, ...]:
+        """Build file transfer jobs."""
+
+        transfer_jobs = []
+
+        for p in source_paths:
+            transfer_jobs.append(
+                TransferJob(
+                    path_src=str(p["spath"]),
+                    path_dest=f"{self.lz_irods_path}/{self.target_coll}/{str(p['ipath'])}",
+                    bytes=p["spath"].stat().st_size,
+                )
+            )
+
+        return tuple(sorted(transfer_jobs))
+
+    def compute_irods_checksums(self, session, jobs: typing.Tuple[TransferJob, ...]):
+        for job in jobs:
+            if not job.path_src.endswith(".md5"):
+                print(
+                    str(
+                        Path(job.path_dest).relative_to(f"{self.lz_irods_path}/{self.target_coll}/")
+                    )
+                )
+                try:
+                    data_object = session.data_objects.get(job.path_dest)
+                    data_object.chksum()
+                except Exception as e:
```

Comment:
This will catch all Exceptions (i.e. network related), not just checksum related ones?